stdlib/public/core/Sequence.swift

```diff
@@ -384,18 +384,27 @@ public protocol Sequence {
     initializing ptr: UnsafeMutableBufferPointer<Element>
   ) -> (Iterator,UnsafeMutableBufferPointer<Element>.Index)
   
-  /// Call `body(p)`, where `p` is a pointer to the collection's
-  /// contiguous storage.  If no such storage exists, it is
+  /// Call `body(buffer)`, where `buffer` is a buffer to the contiguous
+  /// storage of the entire collection. If no such storage exists, it is
   /// first created.  If the collection does not support an internal
   /// representation in a form of contiguous storage, `body` is not
   /// called and `nil` is returned.
   ///
+  /// The optimizer can often eliminate bounds- and uniqueness-checking
+  /// within an algorithm. When that fails, however, invoking the same
+  /// algorithm on `body`\ 's argument may let you trade safety for speed.
+  ///
   /// A `Collection` that provides its own implementation of this method
-  /// must also guarantee that an equivalent buffer of its `SubSequence` 
-  /// can be generated by advancing the pointer by the distance to the
-  /// slice's `startIndex`.
+  /// must guarantee that contiguous storage to any valid `SubSequence`
+  /// can be generated by slicing `buffer` with a range formed from the
+  /// distances to the subsequence's `startIndex` and `endIndex`, respectively.
```

Comment:
This passage is a definite improvement, but I find it a bit difficult to understand what it means. Is it equivalent to stating that the order of elements in the buffer must match the order they appear in the collection?

Reply:
I think it does. That would be a good way to say it.